src/containers/settings/data/processing.js

```diff
@@ -0,0 +1,222 @@
+    React.PropTypes.shape({
+      name: React.PropTypes.string.isRequired,
+      value: React.PropTypes.arrayOf(
+        React.PropTypes.shape({
+          start: React.PropTypes.number.isRequired,
+          rate: React.PropTypes.number.isRequired,
+        }),
+      ),
+    }),
+  ).isRequired,
+  bgTargets: React.PropTypes.objectOf(
+    React.PropTypes.arrayOf(
+      React.PropTypes.shape({
+        start: React.PropTypes.number.isRequired,
+        target: React.PropTypes.number.isRequired,
+      })
+    ),
+  ).isRequired,
+  carbRatios: React.PropTypes.objectOf(
+    React.PropTypes.arrayOf(
+      React.PropTypes.shape({
+        start: React.PropTypes.number.isRequired,
+        amount: React.PropTypes.number.isRequired,
+      })
+    ),
+  ).isRequired,
+  insulinSensitivities: React.PropTypes.objectOf(
+    React.PropTypes.arrayOf(
+      React.PropTypes.shape({
+        start: React.PropTypes.number.isRequired,
+        amount: React.PropTypes.number.isRequired,
+      })
+    ),
+  ).isRequired,
+});
+
+export function processBasalRateData(scheduleData) {
+  const starts = scheduleData.value.map(s => s.start);
+  const noData = [{ start: '-', rate: '-' }];
+
+  if (starts.length === 0) {
+    return noData;
+  } else if (starts.length === 1) {
+    if (Number(utilities.getBasalRate(scheduleData.value, starts[0])) === 0) {
+      return noData;
+    }
+  }
+
+  const data = starts.map((startTime) => ({
+    start: utilities.getFormattedTime(
+      startTime
+    ),
+    rate: utilities.getBasalRate(
+      scheduleData.value,
+      startTime
+    ),
+  }));
+
+  data.push({
+    start: 'Total',
+    rate: utilities.getTotalBasalRates(scheduleData.value),
+  });
+
+  return data;
+}
+
+export function processBgTargetData(targetsData, bgUnits, keys) {
+  const starts = targetsData.map(s => s.start);
+
+  return starts.map((startTime) => ({
+    start: utilities.getFormattedTime(
+      startTime
+    ),
+    columnTwo: utilities.getBloodGlucoseValue(
+      targetsData,
+      keys.columnTwo,
+      startTime,
+      bgUnits
+    ),
+    columnThree: utilities.getBloodGlucoseValue(
+      targetsData,
+      keys.columnThree,
+      startTime,
+      bgUnits
+    ),
+  }));
+}
+
+export function processCarbRatioData(carbRatioData) {
+  const starts = carbRatioData.map(s => s.start);
+  return starts.map((startTime) => ({
+    start: utilities.getFormattedTime(
+      startTime
+    ),
+    amount: utilities.getValue(
+      carbRatioData,
+      'amount',
+      startTime
+    ),
+  }));
+}
+
+export function processSensitivityData(sensitivityData, bgUnits) {
+  const starts = sensitivityData.map(s => s.start);
+  return starts.map((startTime) => ({
+    start: utilities.getFormattedTime(
+      startTime
+    ),
+    amount: utilities.getBloodGlucoseValue(
+      sensitivityData,
+      'amount',
+      startTime,
+      bgUnits
+    ),
+  }));
+}
+
+export function processTimedSettings(pumpSettings, schedule, bgUnits) {
```

Comment:
So since this one is totally specific to Tandem, I personally would probably just keep it in the Tandem.js component. I tend to feel only shared things belong factored out. Thoughts?